### DISCUSSION TITLE
Which class or function are responsible for excecute commands ?

### DISCUSSION THREAD

**Discussion by gorgeously-app** [termux/termux-app#2457] [Q&A] [answered]
I'm thinking about InputText but couldn't find the correct one. Can you someone help me out ? Thanks !

**Comment by agnostic-apollo** [comment, marked as answer]
Commands run with which `API`?

The [`TermuxService`](https://github.com/termux/termux-app/blob/v0.117/app/src/main/java/com/termux/app/TermuxService.java) receives execution commands from plugins and other APIs and runs them with [`TermuxSession`](https://github.com/termux/termux-app/blob/v0.117/termux-shared/src/main/java/com/termux/shared/shell/TermuxSession.java) and  [`TermuxTask`](https://github.com/termux/termux-app/blob/v0.117/termux-shared/src/main/java/com/termux/shared/shell/TermuxTask.java).

**Reply by gorgeously-app** [reply]
Sorry but I couldn't find it in these classes, can you please point out directly which function ? It seems like Runtime.exec but I'm not sure.

**Reply by agnostic-apollo** [reply]
For background shell tasks for termux plugins and `RUN_COMMAND` intent, `Runtime.exec()` is used via `TermuxTask`.
